nodebuilder/blobstream/blobstream.go

```diff
@@ -0,0 +1,95 @@
+package blobstream
+
+import (
+	"context"
+
+	"github.com/celestiaorg/celestia-node/share"
+)
+
+var _ Module = (*API)(nil)
+
+// Module defines the API related to interacting with the proofs
+//
+//go:generate mockgen -destination=mocks/api.go -package=mocks . Module
+type Module interface {
+	// DataCommitment collects the data roots over a provided ordered range of blocks,
+	// and then creates a new Merkle root of those data roots. The range is end exclusive.
+	DataCommitment(ctx context.Context, start, end uint64) (*ResultDataCommitment, error)
+
+	// DataRootInclusionProof creates an inclusion proof for the data root of block
+	// height `height` in the set of blocks defined by `start` and `end`. The range
+	// is end exclusive.
+	DataRootInclusionProof(
+		ctx context.Context,
+		height int64,
+		start, end uint64,
+	) (*ResultDataRootInclusionProof, error)
+
+	// ProveShares generates a share proof for a share range.
+	ProveShares(ctx context.Context, height, start, end uint64) (*ResultShareProof, error)
+	// ProveCommitment generates a commitment proof for a share commitment.
+	ProveCommitment(
+		ctx context.Context,
+		height uint64,
+		namespace share.Namespace,
+		shareCommitment []byte,
+	) (*ResultCommitmentProof, error)
+}
+
+type Internal struct {
```

Comment:
In other files we keep `Interal` as an embedded struct field. Not sure how important it's, commenting just for the consistency 🤷

Reply:
[f8a3066](https://github.com/celestiaorg/celestia-node/pull/3470/commits/f8a30668432f6aae6254a04f9b552b7f3ef5a36a)